crates/sparrow-compiler/src/ast_to_dfg.rs

```diff
@@ -517,15 +517,28 @@ pub fn add_to_dfg(
                 .try_collect()?;
 
             let args: Vec<_> = if function.is_aggregation() {
-                // If the function is an aggregation, we may need to flatten the window.
-                dfg.enter_env();
-                dfg.bind("$condition_input", args[0].inner().clone());
-
-                let window = &expr.unwrap().args()[1];
-                let (condition, duration) =
-                    flatten_window_args_if_needed(window, dfg, data_context, diagnostics)?;
+                let window_arg = original_ast.map(|e| &e.args()[1]);
+                let (condition, duration) = match window_arg {
+                    Some(window) => {
+                        // If the function is an aggregation, we may need to flatten the window.
+                        dfg.enter_env();
+                        dfg.bind("$condition_input", args[0].inner().clone());
+
+                        let result =
+                            flatten_window_args_if_needed(window, dfg, data_context, diagnostics)?;
+                        dfg.exit_env();
+                        result
+                    }
+                    None => {
+                        // If `expr` is None, we're running the Python builder code,
+                        // which already flattened things.
+                        //
+                        // Note that this won't define the `condition_input` for the
+                        // purposes of ticks.
+                        (args[1].clone(), args[2].clone())
```

Comment:
The difference here is that `args[1]` is already a boolean, not a window. That's why the change to the `dfg_signature` works with the python test, and not sparrow. 

Which is somewhat interesting -- with some refactoring to the `ast_to_dfg`, we could probably support `window=daily()` instead of `window=since(daily())`.

Reply:
This should work after fixing my signature bugs in https://github.com/kaskada-ai/kaskada/pull/587